vividus-plugin-web-app/src/main/java/org/vividus/proxy/steps/ProxySteps.java

```diff
@@ -238,6 +243,32 @@ public String toString()
         });
     }
 
+    /**
+     * Add headers to proxy request that will be used while sending request with given urlPattern
+     * @param headers ExamplesTable representing list of headers with columns "name" and "value" specifying HTTP header
+     * names and values respectively
+     * @param comparisonRule String comparison rule: "is equal to", "contains", "does not contain"
+     * @param urlPattern The string value of URL-pattern to filter by
+     */
+    @When("I add headers to proxied requests with URL pattern which $comparisonRule `$urlPattern`:$headers")
+    public void addHeadersToProxyRequest(StringComparisonRule comparisonRule, String urlPattern, ExamplesTable headers)
+    {
+        DefaultHttpHeaders httpHeaders = new DefaultHttpHeaders();
+        headers.getRowsAsParameters(true)
+                .forEach(row -> httpHeaders.add(
+                        row.valueAs("name", String.class),
+                        (Object) row.valueAs("value", String.class)
+                ));
+        Matcher<String> expected = comparisonRule.createMatcher(urlPattern);
+        proxy.addRequestFilter((request, contents, messageInfo) -> {
```

Comment:
@valfirst  in the current implementation, the filter will be cleared off on before scenario hook.
Is it expected behavior?
Shouldn't we support filters that will work whole run time? (For example, if we need to pass basic authentication for some of the requests)
And I think we should give the user the possibility to clear the filters via step.

Reply:
that's a good point, but it's out of scope this PR